File: src/Features.CSharp/CollectionsModel.cs
using System.Collections.Generic;

namespace Features
{
    public class CollectionsModel
    {
        private readonly List<Contact> _contacts = new List<Contact>
        {
            new Contact
            {
                Name = "Paul Knopf",
                PhoneNumber = "525-525-52555"
            }
        };

        public bool AddContact(string name, string phoneNumber)
        {
            if (string.IsNullOrEmpty(name) || string.IsNullOrEmpty(phoneNumber))
            {
                return false;
            }
            
            _contacts.Add(new Contact
                {
                    Name = name,
                    PhoneNumber = phoneNumber
                });
            
            return true;
        }

        public void RemoveContact(int index)
        {
            _contacts.RemoveAt(index);
        }

        public List<Contact> Contacts
        {
            get => _contacts;
        }

        public class Contact
        {
            public string Name { get; set; }
            
            public string PhoneNumber { get; set; }
        }
    }
}

File: src/Features.FSharp/CollectionsModel.fs
namespace Features

open System.Collections.Generic

type Contact = { Name: string; PhoneNumber: string }

type CollectionsModel() =
    let mutable contacts = List<Contact> [ {Name = "Paul Knopf"; PhoneNumber = "525-525-52555" } ]

    member this.AddContact(name: string, phoneNumber: string) =
        if System.String.IsNullOrEmpty name || System.String.IsNullOrEmpty phoneNumber then false
        else 
            contacts.Add {Name = name; PhoneNumber = phoneNumber}
            true

    member this.RemoveContact(index: int) = contacts.RemoveAt(index)
    
    member this.Contacts with get() = contacts
